operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -203,48 +203,97 @@ func (r *RedpandaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
+		// we might have no brokers ready at this point, so we can't do anything below this
+		// since they require the ability to talk to our brokers
 		return r.syncStatusAndRequeue(ctx, status, rp)
 	}
-	defer admin.Close()
-
-	status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonSynced)
 
-	version, requeue, err := r.reconcileClusterConfig(ctx, admin, rp)
+	admin, err := r.ClientFactory.RedpandaAdminClient(ctx, rp)
 	if err != nil {
-		status.Status.SetConfigurationApplied(statuses.ClusterConfigurationAppliedReasonError, err.Error())
-
-		log.Error(err, "error reconciling cluster config")
+		log.Error(err, "error fetching redpanda admin client")
 		return r.syncStatusErr(ctx, err, status, rp)
 	}
+	defer admin.Close()
 
-	status.ConfigVersion = ptr.To(version)
+	// now we ensure that we reconcile all of our decommissioning nodes
+	// TODO: Do we want to rate limit this as well given that it also calls the admin API?
+	// My thought is no since we want to be snappy with decommissioning.
+	health, requeue, err := r.reconcileDecommission(ctx, rp, admin, pools)
+	if err != nil {
+		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonError, err.Error())
+		if err != nil {
+			if internalclient.IsTerminalClientError(err) {
+				status.Status.SetHealthy(statuses.ClusterHealthyReasonTerminalError, err.Error())
+			} else {
+				status.Status.SetHealthy(statuses.ClusterHealthyReasonError, err.Error())
+			}
+		}
 
+		log.Error(err, "error decommissioning brokers")
+		return r.syncStatusErr(ctx, err, status, rp)
+	}
 	if requeue {
 		return r.syncStatusAndRequeue(ctx, status, rp)
 	}
 
-	status.Status.SetConfigurationApplied(statuses.ClusterConfigurationAppliedReasonApplied)
-
-	license, err := r.reconcileLicense(ctx, admin, rp, status)
-	if err != nil {
-		log.Error(err, "error reconciling license")
-		return r.syncStatusErr(ctx, err, status, rp)
+	status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonSynced)
+	if health.IsHealthy {
+		status.Status.SetHealthy(statuses.ClusterHealthyReasonHealthy)
+	} else {
+		// TODO: give more specific message here
+		status.Status.SetHealthy(statuses.ClusterHealthyReasonNotHealthy, "Cluster is not healthy")
 	}
-	if license != nil {
-		rp.Status.LicenseStatus = license
+
+	// we rate limit setting cluster configuration to once a minute if it's already been applied and is up-to-date
+	// NB: For this and the next block, we heavily rely on the likelihood of things getting requeued on a fairly regular
+	// basis, but in some odd cases this could potentially be problematic. Take for instance someone changing a ConfigMap
+	// that gets materialized into the cluster configuration twice in somewhat rapid succession. Let's say that between changes
+	// reconciliation gets run and configuration is synced. After the second change, if reconciliation is triggered again, then
+	// the underlying configuration will not wind up getting updated because:
```

Comment:
Think this is worth putting into an FAQ for the docs?

Reply:
Maybe, but might want to work with @JakeSCahill on how to express it in a way that's digestible and not too low-level. I'm fairly certain that this, and some other edge cases we talked about in Slack, is how our 2.3.x and 2.4.x reconcilers work as well, so it's _probably_ worth calling out sooner than later... but just wondering how we'd want to surface this in a way that makes sense to users...